src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -160,6 +146,21 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
         }
     }
 
+    private void OnWorkspaceChangedImmediate(object? sender, WorkspaceChangeEventArgs e)
+    {
+        if (e.Kind == WorkspaceChangeKind.DocumentChanged)
+        {
+            var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
+            var newDocument = e.NewSolution.GetDocument(e.DocumentId);
+
+            Debug.Assert(oldDocument != null && newDocument != null);
```

Comment:
```suggestion
            var oldDocument = e.OldSolution.GetRequiredDocument(e.DocumentId);
            var newDocument = e.NewSolution.GetRequiredDocument(e.DocumentId);
```

If we're wrong, telemetry will tell us since you're adding that catch-and-report. :smile: